lib/inc/facter/ruby/ruby.hpp

```diff
@@ -41,6 +42,16 @@ namespace facter { namespace ruby {
     LIBFACTER_EXPORT void load_custom_facts(facter::facts::collection& facts, std::vector<std::string> const& paths = {});
 
     /**
+     * Traverses a ruby fact and returns a new value based on the
+     * query segments passed in the range.
+     * @param value The original value to query
+     * @param segment The beginning of the query segment range
+     * @param end The end of the query segment range
+     * @return Returns a pointer to the value queried, or nullptr if it does not exist.
+     */
+    LIBFACTER_EXPORT facts::value const* lookup(facts::value const* value, std::vector<std::string>::iterator segment, std::vector<std::string>::iterator end);
```

Comment:
I think we can live with this for now rather than having to shuffle around headers and what not.